extractor/filesystem/os/pacman/pacman.go

```diff
@@ -0,0 +1,273 @@
+
+	if !strings.HasPrefix(path, archPrefix) || !strings.HasSuffix(path, archSuffix) {
+		return false
+	}
+
+	fileinfo, err := api.Stat()
+	if err != nil {
+		return false
+	}
+	if e.maxFileSizeBytes > 0 && fileinfo.Size() > e.maxFileSizeBytes {
+		e.reportFileRequired(path, fileinfo.Size(), stats.FileRequiredResultSizeLimitExceeded)
+		return false
+	}
+
+	e.reportFileRequired(path, fileinfo.Size(), stats.FileRequiredResultOK)
+	return true
+}
+
+func (e Extractor) reportFileRequired(path string, fileSizeBytes int64, result stats.FileRequiredResult) {
+	if e.stats == nil {
+		return
+	}
+	e.stats.AfterFileRequired(e.Name(), &stats.FileRequiredStats{
+		Path:          path,
+		Result:        result,
+		FileSizeBytes: fileSizeBytes,
+	})
+}
+
+// Extract extracts packages from "desc" files passed through the scan input.
+func (e Extractor) Extract(ctx context.Context, input *filesystem.ScanInput) ([]*extractor.Inventory, error) {
+	inventory, err := e.extractFromInput(ctx, input)
+
+	if e.stats != nil {
+		var fileSizeBytes int64
+		if input.Info != nil {
+			fileSizeBytes = input.Info.Size()
+		}
+		e.stats.AfterFileExtracted(e.Name(), &stats.FileExtractedStats{
+			Path:          input.Path,
+			Result:        filesystem.ExtractorErrorToFileExtractedResult(err),
+			FileSizeBytes: fileSizeBytes,
+		})
+	}
+	return inventory, err
+}
+
+func (e Extractor) extractFromInput(ctx context.Context, input *filesystem.ScanInput) ([]*extractor.Inventory, error) {
+	m, err := osrelease.GetOSRelease(input.FS)
+	if err != nil {
+		log.Errorf("osrelease.ParseOsRelease(): %v", err)
+	}
+
+	s := bufio.NewScanner(input.Reader)
+	var pkgName, pkgVersion, pkgDescription, pkgDependencies string
+	pkgs := []*extractor.Inventory{}
+
+	for s.Scan() {
+		line := s.Text()
+		line = strings.TrimSpace(line)
+
+		if len(line) == 0 {
+			continue
+		}
+
+		if strings.HasPrefix(line, "%NAME%") {
+			pkgName = extractValue(s)
+		} else if strings.HasPrefix(line, "%VERSION%") {
+			pkgVersion = extractValue(s)
+		} else if strings.HasPrefix(line, "%DESC%") {
+			pkgDescription = extractValue(s)
+		} else if strings.HasPrefix(line, "%DEPENDS%") {
+			pkgDependencies = extractValues(s)
+		}
+	}
+
+	i := &extractor.Inventory{
+		Name:    pkgName,
+		Version: pkgVersion,
+		Metadata: &Metadata{
+			PackageName:    pkgName,
+			PackageVersion: pkgVersion,
+			OSID:           m["ID"],
+			OSVersionID:    m["VERSION_ID"],
+		},
+		Locations: []string{input.Path},
+	}
+
+	if pkgDescription != "" {
+		i.Metadata.(*Metadata).PackageDescription = pkgDescription
+	}
+	if len(pkgDependencies) != 0 {
+
+		i.Metadata.(*Metadata).PackageDependencies = pkgDependencies
+	}
+
+	pkgs = append(pkgs, i)
+
+	return pkgs, nil
+}
+
+func extractValue(scanner *bufio.Scanner) string {
+	// Skip the current line.
+	scanner.Scan()
+
+	return strings.TrimSpace(scanner.Text())
+}
+
+func extractValues(scanner *bufio.Scanner) string {
+	var values []string
+
+	// Skip the current line.
+	scanner.Scan()
```

Comment:
We shouldn't skip this else we'll skip the first dependency
e.g. for 
```
%DEPENDS%
sh
glibc
mpfr
```

The scanner would currently find `glibc, mpfr` instead of `sh, glibc, mpfr`.